lib/hammer_cli/subcommand.rb

```diff
@@ -59,23 +97,10 @@ def subcommand(name, description, subcommand_class = self, &block)
             :new_class => subcommand_class
           }
         end
-        super
+        subcommand_class = Class.new(subcommand_class, &block) if block
+        declare_subcommand_parameters unless has_subcommands?
```

Comment:
This line breaks systest: https://ci.theforeman.org/job/systest_foreman/13810/console

```
not ok 16 check smart proxy is registered and Hammer runs
# (in test file /usr/bin/fb-install-foreman.bats, line 167)
#   `count=$(hammer $(tHammerCredentials) --csv proxy list | wc -l)' failed
# /opt/theforeman/tfm/root/usr/share/gems/gems/hammer_cli-0.12.pre.develop/lib/hammer_cli/subcommand.rb:101:in `define_subcommand': undefined local variable or method `declare_subcommand_parameters' for HammerCLI::DefaultsCommand:Class (NameError)
# 	from /opt/theforeman/tfm/root/usr/share/gems/gems/hammer_cli-0.12.pre.develop/lib/hammer_cli/subcommand.rb:77:in `subcommand'
# 	from /opt/theforeman/tfm/root/usr/share/gems/gems/hammer_cli-0.12.pre.develop/lib/hammer_cli/abstract.rb:212:in `block in autoload_subcommands'
# 	from /opt/theforeman/tfm/root/usr/share/gems/gems/hammer_cli-0.12.pre.develop/lib/hammer_cli/abstract.rb:211:in `each'
# 	from /opt/theforeman/tfm/root/usr/share/gems/gems/hammer_cli-0.12.pre.develop/lib/hammer_cli/abstract.rb:211:in `autoload_subcommands'
# 	from /opt/theforeman/tfm/root/usr/share/gems/gems/hammer_cli-0.12.pre.develop/lib/hammer_cli/defaults_commands.rb:158:in `<class:DefaultsCommand>'
# 	from /opt/theforeman/tfm/root/usr/share/gems/gems/hammer_cli-0.12.pre.develop/lib/hammer_cli/defaults_commands.rb:22:in `<module:HammerCLI>'
# 	from /opt/theforeman/tfm/root/usr/share/gems/gems/hammer_cli-0.12.pre.develop/lib/hammer_cli/defaults_commands.rb:3:in `<top (required)>'
# 	from /opt/rh/rh-ruby22/root/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54:in `require'
# 	from /opt/rh/rh-ruby22/root/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54:in `require'
# 	from /opt/theforeman/tfm/root/usr/share/gems/gems/hammer_cli-0.12.pre.develop/lib/hammer_cli/defaults.rb:1:in `<top (required)>'
# 	from /opt/rh/rh-ruby22/root/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54:in `require'
# 	from /opt/rh/rh-ruby22/root/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54:in `require'
# 	from /opt/theforeman/tfm/root/usr/share/gems/gems/hammer_cli-0.12.pre.develop/lib/hammer_cli/context.rb:1:in `<top (required)>'
# 	from /opt/rh/rh-ruby22/root/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54:in `require'
# 	from /opt/rh/rh-ruby22/root/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54:in `require'
# 	from /opt/theforeman/tfm/root/usr/share/gems/gems/hammer_cli-0.12.pre.develop/lib/hammer_cli.rb:17:in `<top (required)>'
# 	from /opt/rh/rh-ruby22/root/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54:in `require'
# 	from /opt/rh/rh-ruby22/root/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54:in `require'
# 	from /opt/theforeman/tfm/root/usr/share/gems/gems/hammer_cli-0.12.pre.develop/bin/hammer:117:in `<top (required)>'
# 	from /usr/bin/hammer:23:in `load'
# 	from /usr/bin/hammer:23:in `<main>'
```

Reply:
@ekohl this change require new version of clamp. The packaging PRs are https://github.com/theforeman/foreman-packaging/pull/1872 (RPM) and https://github.com/theforeman/foreman-packaging/pull/1874 (DEB).